src/main/java/com/amihaiemil/zold/RtWallet.java

```diff
@@ -110,11 +113,33 @@ public double balance() throws IOException{
     public void pay(
         final String keygap, final String user,
         final double amount, final String details
-    ) {
-        throw new UnsupportedOperationException(
-            "Not yet implemented. If you can contribute please, do it at "
-            + "https://github.com/amihaiemil/zold-java-client"
+    ) throws IOException {
+        
+        final HttpPost pay = new HttpPost(
+            URI.create(this.baseUri.toString() + "/do-pay")
         );
+        StringBuilder payload = new StringBuilder();
+        payload.append("{");
+        payload.append("\"keygap\":" + keygap + ",");
+        payload.append("\"bnf\":" + user + ",");
+        payload.append("\"amount\":" + amount + ",");
+        payload.append("\"details\":" + details);
+        payload.append("}");
+        pay.setEntity(new StringEntity(payload.toString(),
+             ContentType.APPLICATION_JSON));
+        try {
+            this.client.execute(
+                pay,
+                new ReadString(
```

Comment:
@amihaiemil I know that ReadString doesn't make sense here
We should make something like WaitForPayment?

Reply:
"do-pay" will return jobId and we should wait for it like it we do with "pull" method

Reply:
@amihaiemil We can create WaitForPayment or use WaitFotWallet here too, But we would need to change its name.
What should I go for?

Reply:
I suggest to rename it to WaitForJob. @amihaiemil what do you think?